### PR TITLE
mattermost: 5.15.0 -> 5.20.1

### PR DESCRIPTION
**Pull request by jslight90**
###### Motivation for this change

Update version of Mattermost.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/web-apps/mattermost.nix [hunk 1]
```diff
@@ -207,8 +206,7 @@ in
           PermissionsStartOnly = true;
           User = cfg.user;
           Group = cfg.group;
-          ExecStart = "${pkgs.mattermost}/bin/mattermost" +
-            (lib.optionalString (!cfg.mutableConfig) " -c ${database}");
```
Comment (nh2) [inline nixos/modules/services/web-apps/mattermost.nix:211]: Why is this changing?

Was it wrong before, or is there a change in the verison upgrade that makes it wrong?

Please also write the answer into the commit message.

**Comment by nh2** [comment]
This new version unbreaks new installs for me on 20.03: https://github.com/NixOS/nixpkgs/issues/89194#issuecomment-636238242

**Review by ali-abrar — COMMENTED** [review]
This is below the [minimum version recommended by mattermost](https://docs.mattermost.com/administration/extended-support-release.html).

**Comment by ryantm** [comment]
We're past this version now. If the service needs updates, could you add that in a separate PR?

**Pull request closed without merging**